appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java

```diff
@@ -1080,7 +1082,7 @@ private void deleteSelectedComponent() {
     if (form.getLastSelectedComponent() instanceof MockForm) {
       return;  // Cannot delete MockForm
     }
-    form.getLastSelectedComponent().delete();
+    form.getLastSelectedComponent().openDeleteDialog();
```

Comment:
I'm not sure that it's appropriate to change the semantics of this function.

Reply:
delete function directly deleting the component, without prompt, that's why I created a new function

Reply:
Right, but the issue is that we have this function `deleteSelectedComponent`, which before this change deletes the component, and by making this call here go from `delete` to `openDeleteDialog` now the semantics of the function have changed from delete to open a dialog. You'd need to verify that all call sites for this function expect that behavior (which they probably don't).

Reply:
> Right, but the issue is that we have this function `deleteSelectedComponent`, which before this change deletes the component, and by making this call here go from `delete` to `openDeleteDialog` now the semantics of the function have changed from delete to open a dialog. You'd need to verify that all call sites for this function expect that behavior (which they probably don't).

Ok,  so would it be ok if I create the new function and call openDeleteDialog from that?